sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2594,6 +2601,31 @@ class Analyzer(override val catalogManager: CatalogManager)
         })
     }
 
+    private def resolveTemp(expr: Expression, agg: Aggregate): Expression = {
```

Comment:
This is very similar to the `resolveCol` method inside `resolveExprsWithAggregate`. Can we have a common method to share code?